src/arrayMethodJoin.js

```diff
@@ -4,8 +4,38 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    let trueSeparator;
+
+    if (separator === null) {
+      trueSeparator = 'null';
+    } else {
+      trueSeparator = separator;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let trueItem;
+
+      if (this[i] === null || this[i] === undefined) {
+        trueItem = '';
+      } else {
+        trueItem = this[i];
+      }
+
+      result += trueItem;
```

Comment:
trueItem зайва змінна, попробуй зробити так
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        result += this[i];
      }
```
```